src/lib.rs

```diff
@@ -321,7 +328,7 @@ typed_builder_doc! {
         pub timestamp: Option<OffsetDateTime>,
         #[cfg(feature = "ethers")]
         /// RPC Provider used for on-chain checks. Necessary for contract wallets signatures.
-        pub rpc_provider: Option<Provider<Http>>,
+        pub rpc_provider: Option<Arc<Provider<Http>>>,
```

Comment:
Would you argue that this change is worth breaking the API? Or do you think the dependencies update needs a new major version by itself?

Reply:
I didn't want to put an arc, but the new ethers seems to require it

Reply:
ethers@2 abstracts over middleware ownership. `Contract` is an alias for `ContractInstance<Arc<M>, M>` for packwards compatibility with ethers@1, which would Arc behind the scenes. You want `ContractInstance<B: Borrow<M>, M: Middleware>` which you can instantiate as

```ContractInstance::<&Provider<Http>, Provider<Http>>::new(address, abi, provider);```